lib/query.js

```diff
@@ -2242,6 +2243,11 @@ Query.prototype._find = wrapThunk(function(callback) {
     }
 
     if (!mongooseOptions.populate) {
+      if (mongooseOptions.lean && mongooseOptions.lean.versionKey === false) {
+        docs.forEach((doc) => {
+          delete doc.__v;
```

Comment:
People can set a custom versionKey in their schemas, so we need to cover that scenario as well:
```suggestion
          delete doc.[schema.options.versionKey || '__v'];
```

Reply:
Better approach is to just do `delete doc[schema.options.versionKey]`.

Also need to add a `&& schema.options.versionKey` check on L2246, because setting `schema.options.versionKey = false` is how you disable versioning for the whole schema.